### PR TITLE
Make all callbacks return boolean values

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Updates the `OnSendBlock` and `OnSessionBlock` callbacks to return boolean values which control whether the event/session is sent or not.

## Changeset
- Updated return type of blocks to `BOOL` for both callback types
- Moved `OnSessionBlock` callbacks into the `SessionTracker`. If a callback returns false, the new session is discarded and the original session (if any) is used

Note: the session callback is effectively passed an empty dictionary as its parameter for now, which has been noted as a TODO comment. Future work has been scheduled which will alter the parameter to take a structured class with session information instead.

## Tests
Added new unit tests to verify the behaviour for session blocks.

### CODE REVIEW

File: Source/BugsnagSessionTracker.m [hunk 1]
```diff
@@ -116,11 +117,21 @@ - (void)startNewSessionWithAutoCaptureValue:(BOOL)isAutoCaptured {
...
+                                                       autoCaptured:isAutoCaptured];
+    // TODO JL: refactor to pass as structured data into callback
+    BugsnagSessionTrackingPayload *payload = [[BugsnagSessionTrackingPayload alloc] initWithSessions:@[]
+                                                                                              config:self.config];
+    NSMutableDictionary *data = [payload toJson];
+    for (BugsnagOnSessionBlock cb in self.config.onSessionBlocks) {
+        if (!cb(data)) {
+            return;
+        }
+    }
```
Comment (robinmacharg) [inline Source/BugsnagSessionTracker.m:132]: nit: this refactoring is a good opportunity to improve naming for clarity.  What do you think about:

* `payload` -> `sessions`
* `data` -> `sessionsJson`
* `cb` -> `onSessionBlock`

File: Source/BugsnagSessionTracker.m [hunk 2]
```diff
@@ -116,11 +117,21 @@ - (void)startNewSessionWithAutoCaptureValue:(BOOL)isAutoCaptured {
...
     }
 
-    self.currentSession = [[BugsnagSession alloc] initWithId:[[NSUUID UUID] UUIDString]
-                                                   startDate:[NSDate date]
-                                                        user:self.config.user
-                                                autoCaptured:isAutoCaptured];
+    BugsnagSession *newSession = [[BugsnagSession alloc] initWithId:[[NSUUID UUID] UUIDString]
+                                                          startDate:[NSDate date]
+                                                               user:self.config.user
+                                                       autoCaptured:isAutoCaptured];
```
Comment (robinmacharg) [inline Source/BugsnagSessionTracker.m:123]: This has become separated from its first use.

File: Source/BugsnagSessionTracker.m (hunk 1 above)
Comment (fractalwrench) [reply Source/BugsnagSessionTracker.m:132]: sounds good :) renamed

File: Source/BugsnagSessionTracker.m (hunk 2 above)
Comment (fractalwrench) [reply Source/BugsnagSessionTracker.m]: Move closer to first assignment.

**Pull request merged**